ecosystem/sf-indexer/aptos-sf-indexer/migrations/2022-08-08-043603_core_tables/up.sql

```diff
@@ -0,0 +1,285 @@
+ "signature":"0x70781112e78cc8b54b86805c016cef2478bccdef21b721542af0323276ab906c989172adffed5bf2f475f2ec3a5b284a0ac46a6aef0d79f0dbb6b85bfca0080a"
+ },
+ "events":[
+ {
+ "key":"0x040000000000000000000000000000000000000000000000000000000000000000000000fefefefe",
+ "sequence_number":"0",
+ "type":"0x1::Whatever::FakeEvent1",
+ "data":{
+ "amazing":"1"
+ }
+ },
+ {
+ "key":"0x040000000000000000000000000000000000000000000000000000000000000000000000fefefefe",
+ "sequence_number":"1",
+ "type":"0x1::Whatever::FakeEvent2",
+ "data":{
+ "amazing":"2"
+ }
+ }
+ ],
+ "timestamp":"1649713141723410"
+ }
+ */
+CREATE TABLE user_transactions (
+  version BIGINT UNIQUE PRIMARY KEY NOT NULL,
+  block_height BIGINT NOT NULL,
+  parent_signature_type VARCHAR(255) NOT NULL,
+  sender VARCHAR(255) NOT NULL,
+  sequence_number BIGINT NOT NULL,
+  max_gas_amount NUMERIC NOT NULL,
+  expiration_timestamp_secs TIMESTAMP NOT NULL,
+  gas_unit_price NUMERIC NOT NULL,
+  -- from UserTransaction
+  "timestamp" TIMESTAMP NOT NULL,
+  -- Default time columns
+  inserted_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  -- Constraints
+  CONSTRAINT fk_versions FOREIGN KEY (version) REFERENCES transactions (version),
+  UNIQUE (sender, sequence_number)
+);
+CREATE INDEX ut_sender_index ON user_transactions (sender);
+CREATE TABLE signatures (
+  transaction_version BIGINT NOT NULL,
+  multi_agent_index BIGINT NOT NULL,
+  multi_sig_index BIGINT NOT NULL,
+  transaction_block_height BIGINT NOT NULL,
+  signer VARCHAR(255) NOT NULL,
+  is_sender_primary BOOLEAN NOT NULL,
+  type VARCHAR(255) NOT NULL,
+  public_key VARCHAR(255) NOT NULL,
+  threshold BIGINT NOT NULL,
+  public_key_indices jsonb NOT NULL,
+  inserted_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  -- Constraints
+  PRIMARY KEY (
+    transaction_version,
+    multi_agent_index,
+    multi_sig_index
+  ),
+  CONSTRAINT fk_transaction_versions FOREIGN KEY (transaction_version) REFERENCES transactions (version)
+);
+/** Ex:
+ {
+ "key": "0x0400000000000000000000000000000000000000000000000000000000000000000000000a550c18",
+ "sequence_number": "0",
+ "type": "0x1::reconfiguration::NewEpochEvent",
+ "data": {
+ "epoch": "1"
+ }
+ }
+ */
+CREATE TABLE events (
+  key VARCHAR(100) NOT NULL,
+  sequence_number BIGINT NOT NULL,
+  creation_number BIGINT NOT NULL,
+  account_address VARCHAR(100) NOT NULL,
+  transaction_version BIGINT NOT NULL,
+  transaction_block_height BIGINT NOT NULL,
+  type TEXT NOT NULL,
+  data jsonb NOT NULL,
+  inserted_at TIMESTAMP NOT NULL DEFAULT NOW(),
+  -- Constraints
+  PRIMARY KEY (key, sequence_number),
+  CONSTRAINT fk_transaction_versions FOREIGN KEY (transaction_version) REFERENCES transactions (version)
+);
+CREATE INDEX event_key_seq_type_index ON events (key, sequence_number, type);
+CREATE TABLE write_set_changes (
+  transaction_version BIGINT NOT NULL,
+  hash VARCHAR(255) NOT NULL,
+  transaction_block_height BIGINT NOT NULL,
+  type TEXT NOT NULL,
+  address VARCHAR(255) NOT NULL,
```

Comment:
will probs want indexes on both of these (also address should be limited all across this file to 34)